### ISSUE TITLE
Pathname with or without trailing slashes for directories

### ISSUE THREAD

**Issue by FrankHB** [devkitPro/libfat#7]
Pathname resolution is not conforming to [POSIX](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap04.html#tag_04_11): Trailing slashes lead to problems. The function does not take path "foo" as same as "foo/", which should be all the same to "foo/." according to the specification. This causes functions like `opendir` to fail when the parameter has no slash(es) at end.

This should be fixed, or at least documented.

**Comment by WinterMute** [comment]
what does opendir("C:") do on windows, do you know offhand. Does it open root or does it open current working directory.

**Comment by FrankHB** [comment]
As I tested with x86_64-w64-mingw32-gcc (from MSYS2) in Windows 10, `opendir("C:")` opens the root directory of drive "C:".

Note that x86_64-pc-msys (also from MSYS2)  opens the root directory successfully with `opendir("C:\\")` but failed with `opendir("C:")`. This is expected because MSYS2 programs as well as Cygwin programs need POSIX-style paths except for several cases (including `C:\` but not `C:`)where they can be recognized and converted by the MSYS/Cygwin runtime to the POSIX paths.

**Comment by WinterMute** [comment]
Did you try chdir before an opendir though?

**Comment by WinterMute** [comment]
Yeah, thought so. I just checked this and the behaviour is what I expected.

On windows opendir("c:") will open the current working directory on the drive, not the root directory. You get the same sort of behaviour with chdir too in that chdir("c:") will change working directory to the last working directory set on drive C. This isn't something I really considered supporting in libfat since, for most purposes, there's only a single storage device.

Being honest it was never intended that code using libfat should ever use the pseudo drive specifier explicitly in any path. That's obviously not portable.

I'll have a look at the trailing slash thing though.

**Comment by FrankHB** [comment]
Sorry, my mistake. The tests were run weeks ago, and I did not call `chdir` then. It indeed opens the current working directory.

Thank your for clarifying the support status.

As the drive specifier, it is indeed not portable in hard-coded arguments. But it can still be portable in a limited form using string concatenation, provided routines to get different path prefix (maybe at runtime) for different platforms.
